lib/rules/no-warning-comments.js

```diff
@@ -64,59 +64,36 @@ module.exports = {
          */
         function convertToRegExp(term) {
             const escaped = escapeRegExp(term);
-            const wordBoundary = "\\b";
-            const eitherOrWordBoundary = `|${wordBoundary}`;
-            let prefix;
 
             /*
-             * If the term ends in a word character (a-z0-9_), ensure a word
-             * boundary at the end, so that substrings do not get falsely
-             * matched. eg "todo" in a string such as "mastodon".
-             * If the term ends in a non-word character, then \b won't match on
-             * the boundary to the next non-word character, which would likely
-             * be a space. For example `/\bFIX!\b/.test('FIX! blah') === false`.
-             * In these cases, use no bounding match. Same applies for the
-             * prefix, handled below.
+             * When matching at the start, ignore leading whitespace, and
+             * there's no need to worry about word boundaries.
+             *
+             * These expressions for the prefix and suffix are designed as follows:
+             * ^   handles any terms at the beginning of a line
+             *     e.g. terms ["TODO"] matches `//TODO something`
+             * $   handles any terms at the end of a line
+             *     e.g. terms ["TODO"] matches `// something TODO`
```

Comment:
```suggestion
             * ^   handles any terms at the beginning
             *     e.g. terms ["TODO"] matches `//TODO something`
             * $   handles any terms at the end
             *     e.g. terms ["TODO"] matches `// something TODO`
```

This rule also checks multiline comments (`/* ... */`). If we say "at the beginning/end of a _line_" here, it might suggest that we intended to use the `m` flag.

Reply:
`$` always matches the end of a line, regardless of the multiline flag. So I'll  make that change just for `^`.

Reply:
> `$` always matches the end of a line, regardless of the multiline flag.

Without the multiline flag, it only matches the end of input:

```js
/a$/.test("a"); // true
/a$/.test("a\n"); // false <--
/a$/m.test("a\n"); // true
```

Reply:
oh, you're right. I even tested that to make sure. I don't know what I did to not get that result. 🤦‍♂️